lib/hysync/marc_synchronizer/marc_parsing_methods/project.rb

```diff
@@ -10,22 +12,40 @@ module Project
         def add_project(marc_record, holdings_marc_records, mapping_ruleset)
           existing_project = digital_object_data.fetch('project',{})['string_key']
 
-          project_string_key = nil
-          marc_record.fields('965').each do |field|
-            project_string_key = HYSYNC['project_mappings'][field['a']]
-            break unless project_string_key.nil?
-          end
+          project_string_keys = marc_record.fields('965').map do |field|
+            HYSYNC['project_mappings'][field['a']]
+          end.compact
 
-          if project_string_key.nil?
+          if project_string_keys.empty?
             self.errors << "Could not resolve 965 values to a project for: #{self.clio_id}"
             return
           end
-          return if project_string_key.eql?(existing_project)
-          # TODO: In Hyacinth 3, support multiple project associations
-          raise 'This record already has a project.' unless existing_project.nil?
-          digital_object_data['project'] = {
-            'string_key' => project_string_key
-          }
+          # pop the first mapped project off to see if it's the first or actual project for the object
+          first_project_string_key = project_string_keys.shift
+          if existing_project.nil?
+            digital_object_data['project'] = {
+              'string_key' => first_project_string_key
+            }
+          else
+            # if the existing project doesn't equal the first key, put it all in queue for other_project
+            project_string_keys.unshift(first_project_string_key) unless existing_project.eql?(first_project_string_key)
+          end
+          project_string_keys.each do |project_string_key|
+            next if existing_project.eql?(project_string_key)
+            other_projects = (dynamic_field_data['other_project'] ||= [])
```

Comment:
Can this line be moved up a couple of lines so it's right above the `project_string_keys.each` line, and then it won't re-run during each iteration?

Reply:
Would you want to delete the `'other_project'` key if it's a blank value after? I was trying to avoid dealing with it in what I anticipate is the most common case (only one project).

Reply:
Hmm.  Good question.  If we're okay with allowing voyager data to **remove** a digital object from a project, then we'll want to keep the blank `other_project` key.  Otherwise we can delete the key.

I have a preference for not deleting the key and allowing voyager to control the set of other projects.  We get into trouble when voyager is allowed to set things that can be overridden by non-voyager means later on, so I'd like to give voyager complete ownership of other_project values.

My above argument is starting to make me think again that we may want to allow the voyager sync to change the primary project in Hyacinth 3, if it changes in the MARC data, but that's a bigger and more complicated conversation (for later).

Reply:
Ah, I remember talking about this. The "voyager has total control and deletes" is **not** the spec'd behavior in this PR. I don't think expecting this data to be maintained in Voyager is the right move right now - projects are just Hyacinth data, and there's a lot of movement in the near term between the "RBML General" type projects, Customer Orders, etc. that I don't think we should rely on Voyager for - at least not yet.

Reply:
We can negotiate this later/in a separate ticket.

Reply:
Yeah, agreed.  Not the way to go for now.  Talking later in a separate ticket sounds good!